url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/io/Generic_StaticIO.java

```diff
@@ -0,0 +1,2156 @@
+     *
+     * @param object
+     * @param f
+     */
+    public static void writeObject(
+            Object object,
+            File f) {
+        try {
+            ObjectOutputStream oos;
+            oos = getObjectOutputStream(f);
+            oos.writeObject(object);
+            oos.flush();
+            oos.close();
+        } catch (IOException e) {
+            System.err.print(e.getMessage());
+            e.printStackTrace(System.err);
+            System.exit(Generic_ErrorAndExceptionHandler.IOException);
+        }
+    }
+
+    /**
+     * Read Object from File
+     *
+     * @param f
+     * @return
+     */
+    public static Object readObject(
+            File f) {
+        Object result = null;
+        if (f.length() != 0) {
+            try {
+                ObjectInputStream ois;
+                ois = getObjectInputStream(f);
+                result = ois.readObject();
+                ois.close();
+            } catch (IOException e) {
+                System.err.print(e.getMessage());
+                e.printStackTrace(System.err);
+                System.exit(Generic_ErrorAndExceptionHandler.IOException);
+            } catch (ClassNotFoundException e) {
+                System.err.print(e.getMessage());
+                e.printStackTrace(System.err);
+                System.exit(Generic_ErrorAndExceptionHandler.ClassNotFoundException);
+            }
+        }
+        return result;
+    }
+
+    /**
+     * @param input_File A File which is not a Directory to be copied
+     * @param outputDirectory_File The Directory to copy to.
+     */
+    private static void copyFile(
+            File input_File,
+            File outputDirectory_File) {
+        copyFile(
+                input_File,
+                outputDirectory_File,
+                input_File.getName());
+    }
+
+    /**
+     * @param fileToCopy A File which is not a Directory to be copied
+     * @param directoryToCopyInto A File that is the Directory to copy to.
+     * @param outputFileName The name for the file that will be created in the
+     * outputDirectory_File
+     */
+    public static void copyFile(
+            File fileToCopy,
+            File directoryToCopyInto,
+            String outputFileName) {
+        // String osName = System.getProperty( "os.name" );
+        // if ( osName.equalsIgnoreCase( "UNIX" ) ) {
+        // Runtime.getRuntime().exec( "cp " + _InputFile.toString() + " " +
+        // _Output_File.toString() );
+        // } else {
+        // }
+        if (!fileToCopy.exists()) {
+            System.err.println(
+                    "!input_File.exists() in "
+                    + Generic_StaticIO.class.getCanonicalName()
+                    + ".copy(File(" + fileToCopy + "),File(" + directoryToCopyInto + "))");
+            System.exit(Generic_ErrorAndExceptionHandler.IOException);
+        }
+        if (!directoryToCopyInto.exists()) {
+            directoryToCopyInto.mkdirs();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Confirm that `mkdirs` was successful. Handle any failures that happened. If you fail silently, then you might encounter errors that are harder to debug. See the following diffs on GitHub that make this code change: [diff1](https://github.com/qunarcorp/qmq/commit/12f1d9d6a57a02bc2415cb2af9f4388a77b12c02#diff-6b67be6e17f24fdeddb7c725ac8ec5600d6e63be0bdca0963a83d530e6d55e35L36), [diff2](https://github.com/apache/jmeter/commit/1ec61206779b3f84df18516682b4791bf015178f#diff-cd74304763c097fbc085ac2bc952b46ab3843ac825cacacad89e8064296e955aL330)

Learn more: [File.mkdirs in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#mkdirs--)